red/Animal/lib/Animal.pm

```diff
@@ -32,6 +33,82 @@ This simulates the behaviour of animals.
 
 =head1 SUBROUTINES/METHODS
 
+=head2 new
+
+Spawn an animal with unique features
+
+=cut
+
+sub new {
+  my ( $class, %attr ) = @_;
+
+  my %animal = (
+    name => $attr{name},
+    colour => $attr{colour} // $class->default_colour
+    );
+
+  bless sub {
+    my ( $prop, $val ) = @_;
+
+    if ( !exists $animal{$prop} ) {
+      croak "$class doesn't have property $prop";
+    }
+
+    if ( defined $val ) {
+      $animal{$prop} = $val;
+      return __SUB__;
+    }
+
+    return $animal{$prop};
+  }, $class;
```

Comment:
I think I see what this is doing, but I'm not entirely sure on the mechanics so this is definitely something to go over in the meeting. :horse: